### ISSUE TITLE
Bit creates a bad link-file for a relative import statement from one scss to another (in different components)

### ISSUE THREAD

**Issue by itaymendel** [teambit/bit#2182]
### Describe the bug

I have two components, the first is a React component that has an `scss` file that uses a relative import statement to require another `scss` file. The other `scss` file is tracked as its own component.

When I install the React component the contents of the link file looks like this:

```js
@import '~@bit/<component-namespace>.<component-name>/dist/<main-file>.scss';
```

When I set up a create-react-app to render it I get an error saying it can't resolve the import statement. I'm only able to fix it by editing the `import` statement to:

```js
@import '~@bit/<owner-name>.<collection-name>.<component-namespace>.<component-name>/<main-file>.scss';
```

### Steps to Reproduce

STRs shared privately with @GiladShoham .

### Expected Behavior

Bit should generate the correct `import` statement.

### Specifications

- Bit version: 14.6.0
- Node version: v10.15.0
- npm / yarn version: 6.4.1
- Platform: macos mojave
- Bit compiler (include version): bit.envs/compilers/react-typescript@3.1.11

**Comment by GiladShoham** [comment]
might be related to #1041

**Comment by davidfirst** [comment]
@itaymendel , thanks for the helpful description.

There are two (unrelated) bugs here.
1. the mechanism that changes the dists code to include the scope-name upon built failed. 
That's why the import statement is `'~@bit/<component-namespace>` instead of `'~@bit/<owner-name>.<collection-name>`
The reason is that we look for package name starts with `@bit`, but this one starts with `~@bit`.
That should be an easy fix.
2. the import goes to the `dist/` even though the component doesn't have dist.
After investigation, here is what happened: the first version of the component does include dists. In the second version (0.0.2), the compiler was removed, so no dist. As a result of tagging this dependency component of the scss file, the dependent component was bumped to 0.0.2 as well. 
I'm not sure yet, but I believe that's why the dependent didn't get completely rebuild include its dependencies.

**Comment by GiladShoham** [comment]
@davidfirst sounds like both are important bugs to fix. good catch!

**Comment by davidfirst** [comment]
Both issues described above have fixed, added e2e and unit tests, merged into master.

**Comment by Tallyb** [comment]
@itaymendel  - to be clear: the import SHOULD be from the dist? 
`@import '~@bit/<component-namespace>.<component-name>/dist/<main-file>.scss';`
And putting scss files inside dist does not look very good, either.

**Comment by itaymendel** [comment]
@Tallyb there's no `dist` for the component as there was compiler configured for the component at all.
This was the second issue @davidfirst was referring to.

**Comment by davidfirst** [comment]
Closing the issue as my fixes are part of the newly released version `v14.7.1`.
